### PR TITLE
refactor: use event classes in US-NY parser, remove arrow

### PR DESCRIPTION
**Pull request by yujia21**
## Issue

[6011](https://github.com/electricitymaps/electricitymaps-contrib/issues/6011)

## Description

Use event class, remove arrow library.

### Double check

- [x] I have tested my parser changes locally with `poetry run test_parser "zone_key"`
- [x] I have run `pnpx prettier@2 --write .` and `poetry run format` in the top level directory to format my changes.

### CODE REVIEW

**Review by VIKTORVAV99 — CHANGES_REQUESTED** [review]
Just one small leftover arrow part that needs to be removed as well.

File: parsers/US_NY.py [hunk 1]
```diff
@@ -80,76 +90,68 @@ def data_parser(df) -> list:
...
 
 @refetch_frequency(timedelta(days=1))
 def fetch_production(
-    zone_key: str = "US-NY",
+    zone_key: ZoneKey = ZoneKey(ZONE),
     session: Session | None = None,
     target_datetime: datetime | None = None,
     logger: Logger = getLogger(__name__),
-) -> list:
+) -> list[dict[str, Any]]:
```
[the remark this answers is not in the document]
Comment (yujia21) [reply parsers/US_NY.py:112]: Should we return an empty list instead?

[the remark this answers is not in the document]
Comment (VIKTORVAV99) [reply parsers/US_NY.py:112]: I think that would be best 👍🏼

**Comment by VIKTORVAV99** [comment]
Just to a note for future PRs, please do not use force push. It creates merge conflicts when trying to check out the branch locally which makes it harder to review.

**Review by VIKTORVAV99 — APPROVED** [review]
LGTM!

**Pull request merged**